lib/active_model/serializer.rb

```diff
@@ -111,6 +108,193 @@ def self.serialization_adapter_instance
       @serialization_adapter_instance ||= ActiveModelSerializers::Adapter::Attributes
     end
 
+    # Preferred interface is ActiveModelSerializers.config
+    # BEGIN DEFAULT CONFIGURATION
+    config.collection_serializer = ActiveModel::Serializer::CollectionSerializer
+    config.serializer_lookup_enabled = true
```

Comment:
Noting open issues with this:

- https://github.com/rails-api/active_model_serializers/issues/1500
- https://github.com/rails-api/active_model_serializers/issues/1636
- https://github.com/rails-api/active_model_serializers/issues/2014
- https://github.com/rails-api/active_model_serializers/issues/1813 esp https://github.com/rails-api/active_model_serializers/issues/1813#issuecomment-227226103